main/ZgatewayBT.ino

```diff
@@ -722,6 +722,34 @@ void notifyCB(
       BLEdata.set("volt", (float)(((pData[4] * 256) + pData[3]) / 1000.0));
       BLEdata.set("batt", (float)(((((pData[4] * 256) + pData[3]) / 1000.0) - 2.1) * 100));
 
+      pubBT(BLEdata);
+    } else if (length == 20) {
+      // DT24-BLE data format
+      // https://github.com/NiceLabs/atorch-console/blob/master/docs/protocol-design.md#dc-meter-report
+      // Data comes as two packets ( 20 and 16 ), and am only processing first
+      Log.trace(F("Device identified creating BLE buffer" CR));
+      JsonObject& BLEdata = getBTJsonObject();
+      String mac_adress = pBLERemoteCharacteristic->getRemoteService()->getClient()->getPeerAddress().toString().c_str();
+      mac_adress.toUpperCase();
+      for (vector<BLEdevice*>::iterator it = devices.begin(); it != devices.end(); ++it) {
+        BLEdevice* p = *it;
+        if ((strcmp(p->macAdr, (char*)mac_adress.c_str()) == 0)) {
+          if (p->sensorModel == LYWSD03MMC)
+            BLEdata.set("model", "LYWSD03MMC");
+          else if (p->sensorModel == MHO_C401)
+            BLEdata.set("model", "MHO_C401");
```

Comment:
Not sure you need to do this check, as it is already done before

Reply:
I was looking at the logic in the bluetooth notification callback and was thinking it needs a refactor to handle more than a few devices as the usage of message length to determine the message parser to be used is going to be problematic with more devices.  My thought was that it would make more sense to do the MAC address to sensor model mapping first, then parse based on sensor model.  Or should the logic be length and model ?  Am thinking that this should be done with the next addition.

That IF block is a duplicate of the one when the length is '5' that I augmented with DT24 for length '20'.  Its a little bit redundant but would correctly identify in the case those devices sent a message with length '20'.

Reply:
> My thought was that it would make more sense to do the MAC address to sensor model mapping first

Agreed, it can be done with another PR indeed!

> would correctly identify in the case those devices sent a message with length '20'.

I don't think they do you can remove

Reply:
Added in comment into the code base to reflect this recommendation